dolphin-dnn/src/test/java/edu/snu/dolphin/dnn/NeuralNetworkTest.java

```diff
@@ -15,45 +15,60 @@
  */
 package edu.snu.dolphin.dnn;
 
+import edu.snu.dolphin.dnn.blas.Matrix;
+import edu.snu.dolphin.dnn.blas.MatrixFactory;
+import edu.snu.dolphin.dnn.blas.jblas.MatrixJBLASFactory;
+import edu.snu.dolphin.dnn.blas.MatrixUtils;
 import edu.snu.dolphin.dnn.conf.ActivationLayerConfigurationBuilder;
 import edu.snu.dolphin.dnn.conf.ActivationWithLossLayerConfigurationBuilder;
 import edu.snu.dolphin.dnn.conf.FullyConnectedLayerConfigurationBuilder;
 import edu.snu.dolphin.dnn.conf.NeuralNetworkConfigurationBuilder;
 import edu.snu.dolphin.dnn.layers.LayerParameter;
 import edu.snu.dolphin.dnn.layerparam.provider.LocalNeuralNetParameterProvider;
-import edu.snu.dolphin.dnn.util.Nd4jUtils;
 import org.apache.commons.lang3.ArrayUtils;
 import org.apache.reef.tang.Configuration;
 import org.apache.reef.tang.Injector;
 import org.apache.reef.tang.Tang;
 import org.apache.reef.tang.exceptions.InjectionException;
-import org.nd4j.linalg.api.ndarray.INDArray;
 import org.junit.Before;
 import org.junit.Test;
-import org.nd4j.linalg.factory.Nd4j;
 
 import static org.junit.Assert.*;
 
 /**
  * Unit tests for neural network.
  */
-public class NeuralNetworkTest {
-
-  private final float tolerance = 1e-6f;
+public final class NeuralNetworkTest {
+
+  private static MatrixFactory matrixFactory;
+  private static LayerParameter emptyLayerParam;
+  private static final float TOLERANCE = 1e-7f;
+
+  static {
+    final Configuration configuration = Tang.Factory.getTang().newConfigurationBuilder()
+        .bindImplementation(MatrixFactory.class, MatrixJBLASFactory.class)
+        .build();
+    try {
+      matrixFactory = Tang.Factory.getTang().newInjector(configuration).getInstance(MatrixFactory.class);
+      emptyLayerParam = LayerParameter.newEmptyInstance(matrixFactory);
+    } catch (final InjectionException e) {
+      throw new RuntimeException("InjectionException while injecting a matrix factory: " + e);
+    }
+  }
 
-  private final INDArray input = Nd4j.create(new float[]{77, 57, 30, 26, 75, 74, 87, 75});
-  private final INDArray expectedOutput =
-      Nd4j.create(new float[]{5.96001904648e-01f, 5.54388443935e-01f, 4.88766051729e-01f});
-  private final INDArray label = Nd4j.create(new float[]{0, 1, 0});
+  private final Matrix input = matrixFactory.create(new float[]{77, 57, 30, 26, 75, 74, 87, 75});
+  private final Matrix expectedOutput =
+      matrixFactory.create(new float[]{5.61329743164e-01f, 5.97883503916e-01f, 4.79822915984e-01f});
```

Comment:
While `INDArray` uses a row-major array for storing elements, jblas's `FloatMatrix` uses a column-major array. Despite the same random generator and seed, this difference yields different randomly initialized matrices that represents fully connected layers' weights. This causes different outputs. Thus, I calculated the expected activations, errors and parameters with the new weight matrices using GNU Octave.